vllm/entrypoints/openai/protocol.py

```diff
@@ -567,3 +567,20 @@ class BatchRequestOutput(OpenAIBaseModel):
     # For requests that failed with a non-HTTP error, this will contain more
     # information on the cause of the failure.
     error: Optional[Any]
+
+
+class TokenizeRequest(OpenAIBaseModel):
```

Comment:
Do you think the data being passed should conform to:

- `/completions` endpoint
- `/chat-completions` endpoint

Either way, please make sure that the naming aligns to what is expected for the openai apis. E.g. if we go with `/completions` it should be `prompt`

And I think that the `/tokenize` should do exactly the same preprocessing as we do for the inference endpoints

Reply:
I think it's more conform with the `/completions` endpoint, since tokenization is usually required in lower level applications.

About preprocessing, is this function meant?

https://github.com/vllm-project/vllm/blob/1def8f30dd860de831a3a03258434d11ae6ba56e/vllm/entrypoints/openai/serving_engine.py#L171

If so, I considered this function, but it has unnecessary checks (such as max_model_len), so I decided to use the tokenizer directly. In the application I'm using, `/tokenize` endpoint is used to count tokens and check that they don't exceed the maximum model length before sending a completion request, so the `/tokenize` endpoint completely loses its use in this case.

However, I wouldn't mind adding the parameter to disable these checks.

Reply:
All good - I just meant that whatever we expose should do the same thing

- if conforming to /completions, we should make sure we do the same thing wrt handling special tokens
- if conforming to /chat-completions, we should make sure we apply the chat template in the same way